### PR TITLE
feat(ext/fetch): `Request.bytes()` and `Response.bytes()`

### LINKED ISSUE #23790: Implement Request/Response `.bytes()` method
See https://github.com/whatwg/fetch/issues/1732.

Tests at https://github.com/web-platform-tests/wpt/pull/46198.

The above spec PR adds a `bytes()` method to get a Uint8Array to Body, instead of requiring users to get an ArrayBuffer with `arrayBuffer()` and then wrap it.

There are also PRs to do the same for Blob and PushMessageData, which copy relevant parts of the Body interface:
- https://github.com/w3c/FileAPI/pull/197
- https://github.com/w3c/push-api/pull/369

### PR DESCRIPTION
**Pull request by iuioiua**
Closes #23790

### CODE REVIEW

File: cli/tsc/dts/lib.dom.d.ts [hunk 1]
```diff
@@ -3156,6 +3156,7 @@ interface Body {
     arrayBuffer(): Promise<ArrayBuffer>;
     /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Request/blob) */
     blob(): Promise<Blob>;
+    bytes(): Promise<Uint8Array>;
```
Comment (iuioiua) [inline cli/tsc/dts/lib.dom.d.ts:3159]: This doesn't yet have an MDN reference, but when it does, it'll be at https://developer.mozilla.org/en-US/docs/Web/API/Request/bytes. Is it worth updating now or once it exists?

File: cli/tsc/dts/lib.webworker.d.ts [hunk 2]
```diff
@@ -1029,6 +1029,7 @@ interface Body {
     arrayBuffer(): Promise<ArrayBuffer>;
     /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Request/blob) */
     blob(): Promise<Blob>;
+    bytes(): Promise<Uint8Array>;
```
Comment (iuioiua) [inline cli/tsc/dts/lib.webworker.d.ts:1032]: Ditto

File: ext/fetch/22_body.js [hunk 3]
```diff
@@ -296,6 +296,15 @@ function mixinBody(prototype, bodySymbol, mimeTypeSymbol) {
       configurable: true,
       enumerable: true,
     },
+    bytes: {
+      /** @returns {Promise<Uint8Array>} */
+      value: function bytes() {
+        return consumeBody(this, "bytes");
```
Comment (iuioiua) [inline ext/fetch/22_body.js:302]: Is `bytes` right?

**Comment by iuioiua** [comment]
I updated the `wpt` git submodule but idk why there's a merge conflict. How do I fix I it?

File: cli/tsc/dts/lib.dom.d.ts (hunk 1 above)
Comment (dsherret) [reply cli/tsc/dts/lib.dom.d.ts:3159]: Seems worth adding it now since it will eventually work.

File: ext/fetch/22_body.js (hunk 3 above)
Comment (bartlomieju) [reply ext/fetch/22_body.js:302]: Yep

File: tests/testdata/fmt/with_config/subdir/a.ts [hunk 4]
```diff
@@ -22,15 +22,11 @@ Deno.test(
...
-			)
+			await response
+				.bytes()
 		const u8a1 =
-			new Uint8Array(
-				await response1
-					.arrayBuffer(),
-			)
+			await response1
+				.bytes()
```
Comment (bartlomieju) [inline tests/testdata/fmt/with_config/subdir/a.ts:29]: Nitpick: move these to be single line statements. `deno fmt` preserves line-breaks

File: tests/unit/fetch_test.ts [hunk 5]
```diff
@@ -1214,7 +1214,7 @@ Deno.test(
       ],
       body: stream.readable,
     });
-    await response.arrayBuffer();
+    await response.body?.cancel();
```
Comment (bartlomieju) [inline tests/unit/fetch_test.ts:1217]: Are these `.body?.cancel()` intentional? Why not `.bytes()`?

**Review by bartlomieju — COMMENTED** [review]
@iuioiua please rebase

File: tests/unit/fetch_test.ts (hunk 5 above)
Comment (iuioiua) [reply tests/unit/fetch_test.ts:1217]: `.body?.cancel()` yields the same effect as `.bytes()` and `.arrayBuffer()`, but with less work.

**Review by bartlomieju — APPROVED** [review]

File: tests/testdata/fmt/with_config/subdir/a.ts (hunk 4 above)
Comment (iuioiua) [reply tests/testdata/fmt/with_config/subdir/a.ts:29]: FYI, this test requires the code to look like this.

File: tests/wpt/runner/expectation.json [hunk 6]
```diff
@@ -7153,8 +7153,12 @@
...
         "json.any.html": true,
         "json.any.worker.html": true,
-        "response-blob-realm.any.html": true,
-        "response-blob-realm.any.worker.html": true,
+        "response-blob-realm.any.html": [
+          "realm of the Uint8Array from Response bytes()"
+        ],
+        "response-blob-realm.any.worker.html": [
+          "realm of the Uint8Array from Response bytes()"
+        ],
```
Comment (bartlomieju) [inline tests/wpt/runner/expectation.json:7161]: These look funky. What are these tests doing?

**Review by bartlomieju — CHANGES_REQUESTED** [review]
Temporarily blocking due to unexpected change in WPT

File: tests/wpt/runner/expectation.json (hunk 6 above)
Comment (iuioiua) [reply tests/wpt/runner/expectation.json:7161]: https://github.com/web-platform-tests/wpt/blob/master/fetch/api/response/response-blob-realm.any.js

Comment (bartlomieju) [reply tests/wpt/runner/expectation.json:7161]: Seems fine since we don't support `iframe`.

**Review by bartlomieju — APPROVED** [review]
LGTM https://github.com/denoland/deno/pull/23823#discussion_r1610791607

**Pull request merged**